dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/AliasAnalysis.java

```diff
@@ -41,6 +42,31 @@ static AliasAnalysis fromConfig(Program program, Configuration config) throws In
         return a;
     }
 
+    static boolean virtualLoc(MemEvent e1, MemEvent e2) {
+        if (!(e1.getAddress() instanceof MemoryObject) || !(e2.getAddress() instanceof MemoryObject)) {
+            return false;
+        }
+        MemoryObject add1 = (MemoryObject) e1.getAddress();
+        MemoryObject add2 = (MemoryObject) e2.getAddress();
+        Boolean virtuality1 = add1.getVirtual();
+        Boolean virtuality2 = add2.getVirtual();
+        if (virtuality1 && virtuality2) {
+            // add1, add2 are virtual, and they should alias to the same physical Address
+            return (add1.getAlias() != null && add1.getAlias().equals(add2.getAlias()));
```

Comment:
While physical addresses might not have an alias, AFAIK virtual ones always do (if this is true, you probably want to add a check in the `MemoryObject` constructor in the form of a `Precondition` ... something like ` virtual => alias != null`)

Reply:
Added a Precondition to constructor and set alias to Final.

Reply:
You should better use `checkArgument(!virtual || alias != null)`

Reply:
updated

Reply:
By convention, `checkArgument` is always placed at the beginning of the method (or in this case, the constructor)

Reply:
`checkArgument `has been moved to the beginning of the constructor.